core/domain/user_services.py

```diff
@@ -2171,3 +2178,31 @@ def create_login_url(return_url):
     # flow on top of the Firebase SDK in "core/templates/pages/login-page", and
     # this function will always redirect to its static location ("/login").
     return '/login?%s' % python_utils.url_encode({'return_url': return_url})
+
+
+def update_roles_and_banned_fields(user_settings_model):
+    """Updates the new roles and banned fields in the UserSettingsModel which
+    are not in use but needs to kept in sync.
+
+    TODO(#12755): Remove this function once the roles and banned field of
+    UserSettingsModel are in use. It is not recommended to use this function in
+    new places.
+
+    Args:
+        user_settings_model: UserSettingsModel. The models which needs update.
+    """
+    if user_settings_model.role == feconf.ROLE_ID_BANNED_USER:
+        user_settings_model.banned = True
+        user_settings_model.roles = []
+        return
+    if user_settings_model.role in [
+            feconf.ROLE_ID_LEARNER, feconf.ROLE_ID_EXPLORATION_EDITOR]:
+        user_settings_model.banned = False
+        user_settings_model.roles = [user_settings_model.role]
+        return
+
+    # Learners are not allowed to have other roles, so user with role other than
+    # exploration editor or learner should have exploration editor role.
+    user_settings_model.roles = [
+        feconf.ROLE_ID_EXPLORATION_EDITOR, user_settings_model.role]
+    user_settings_model.banned = False
```

Comment:
I mean if rn someone is an admin they also have the moderator rights, do we want to preserve that?

Reply:
> I mean if rn someone is an admin they also have the moderator rights, do we want to preserve that?

Nope, we are going to change the actions as needed for each role. Currently, it was not possible as we have a hierarchal system.  (Also, we are planning to drop the admin role in the future and add a different role.)

You can find details about all the expected roles and actions (once we start using the roles field) [here](https://docs.google.com/document/d/1xp-HvyVut4ean3p-aAKGQZng0czMYw1K9PHasNg8eXk/edit#heading=h.nef680s0kwxr).